### DISCUSSION TITLE
Import sequelize model into [...nextauth].js

### DISCUSSION THREAD

**Discussion by GodsonAddy** [sequelize/sequelize#15296] [Help & Questions] [answered]
## Issue Creation Checklist
 I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Issue Description

Please I need to know if the approach I have taken is the correct approach. I'm trying to import my model User which I created in my nextjs into my [...nextauth].js. It works fine for now but I want to know if there's a better approach to this. And these are my files.

**models/index.js**
`const fs = require("fs");
const path = require("path");
const Sequelize = require("sequelize");
const process = require("process");
const basename = path.basename(__filename);
const env = process.env.NODE_ENV || "development";
const config = require(__dirname + "/../config/config.json")[env];
const db = {};

const models = process.cwd() + "/db/models/" || __dirname;

let sequelize;
if (config.use_env_variable) {
  sequelize = new Sequelize(process.env[config.use_env_variable], config);
} else {
  sequelize = new Sequelize(
    config.database,
    config.username,
    config.password,
    config
  );
}

fs.readdirSync(__dirname)
  .filter((file) => {
    return (
      file.indexOf(".") !== 0 && file !== basename && file.slice(-3) === ".js"
    );
  })
  .forEach((file) => {
    // const model = require(path.join(__dirname, file))(
    //   sequelize,
    //   Sequelize.DataTypes
    // );

    const model = path.join(models, file);

    db[model.name] = model;
  });

Object.keys(db).forEach((modelName) => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

module.exports = db;
`

**models/user**
`const { DataTypes, Model } = require("sequelize");
const sequelize = require("../../database/index");

class User extends Model {
 
  static associate(models) {
    // define association here
  
  }
}
User.init(
  {
  
    name: DataTypes.STRING,
    email: DataTypes.STRING,
    image: DataTypes.STRING,
      },
  {
    sequelize: sequelize,
    modelName: "User",
    tableName: "users",
  }
);

module.exports = User;`

**database/index**
`import { Sequelize } from "./models/index";

const sequelize = new Sequelize(process.env.DATABASE_URL);

module.exports = sequelize;`

And in my  **[...nextauth].js**, I imported my models User as
`import User from "../../../db/models/user";`

My dialect is postgres

Please this what I have done and any contribution will be welcomed

### Is this issue dialect-specific?

- [ ] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] No, please it's a private repo

**Comment by ephys** [comment, marked as answer]
I'd use esm nowadays. I also wouldn't have a `db` object anymore, I'd just import the models themselves. If you use typescript I'd use sequelize-typescript too

Otherwise it's fine yes

**Reply by GodsonAddy** [reply]
Thanks for the contribution. I'll check the esm also
